### PR TITLE
[WFCORE-3819][WFCORE-3877] added custom-security-event-listener into Elytron

### PR DESCRIPTION
**Pull request by hkalina**
https://issues.jboss.org/browse/WFCORE-3819 (custom audit, https://issues.jboss.org/browse/EAP7-1007, analysis: https://github.com/wildfly/wildfly-proposals/pull/68)
https://issues.jboss.org/browse/WFCORE-3877 (Configurable optional, https://issues.jboss.org/browse/EAP7-1005, analysis: https://github.com/wildfly/wildfly-proposals/pull/69)

test + doc of this: https://github.com/wildfly/wildfly/pull/11367 (https://issues.jboss.org/browse/WFLY-10611)

* added custom-security-event-listener
* reworked CustomComponentDefinition to allow passing wrapper (to allow custom-security-event-listener to implement Consumer<SecurityEvent> and not to depend on subsystem-specific SecurityEventListener)
* as requested by @darranl, removed need to implement Configurable interface requirement - method initialize(Map) sufficient
* in reworked component removed dependencies on deprecated ServiceBuilder methods

### CODE REVIEW

**Comment by hkalina** [comment]
@darranl please review

**Comment by darranl** [comment]
You don't need to worry about WFCORE-3858 being merged first - the purpose of the topic branch is multiple engineers can start from that topic branch and provided no one modifies those commits it does not matter which order their PRs are merged.

In the past we have submitted the version bump in it's own PR but I also don't really like that as if anything happens and none of the modification PRs make it in we then need to revert it.

So overall looks like we need to get it to cover the RFE process.

**Comment by darranl** [comment]
It looks like this is still missing version 4 of the Elytron subsystem model.

**Comment by jmesnil** [comment]
retest this please

**Comment by jmesnil** [comment]
retest this please

**Comment by OndrejKotek** [comment]
LGTM, ok for QE.

**Comment by kabir** [comment]
Readding missing reqs as https://issues.jboss.org/browse/EAP7-1005 is still not there

**Comment by jmesnil** [comment]
i removed the missing-reqs label

File: elytron/src/main/resources/schema/wildfly-elytron_4_0.xsd [hunk 1]
```diff
@@ -497,7 +497,8 @@
             </xs:documentation>
         </xs:annotation>
         <xs:sequence>
-            <xs:element name="aggregate-security-event-listener" type="aggregateSecurityEventListener" minOccurs="0" maxOccurs="unbounded" />
+            <xs:element name="aggregate-security-event-listener" type="aggregateSecurityEventListenerType" minOccurs="0" maxOccurs="unbounded" />
+            <xs:element name="custom-security-event-listener" type="customSecurityEventListenerType" minOccurs="0" maxOccurs="unbounded" />
```
Comment (jmesnil) [inline elytron/src/main/resources/schema/wildfly-elytron_4_0.xsd]: I have not seen such element in XML test to ensure that the parser is correctly parsing them

**Comment by darranl** [comment]
@hkalina please don't rebase this one, if you can add the required test to this branch I can still include this PR in the aggregate PR.

**Comment by hkalina** [comment]
@darranl do you mean to not change the parent, or to not amend commits too?

**Comment by darranl** [comment]
Both please - so just add one new commit to this branch with the amended test.

**Pull request merged**